trace/google/cloud/trace/span_context.py

```diff
@@ -0,0 +1,164 @@
+            span_id=None,
+            enabled=True,
+            from_header=False):
+        if trace_id is None:
+            trace_id = generate_trace_id()
+
+        self.trace_id = self.check_trace_id(trace_id)
+        self.span_id = self.check_span_id(span_id)
+        self.enabled = enabled
+        self.from_header = from_header
+
+    def check_span_id(self, span_id):
+        """Check the type of span_id to ensure it is int. If it is not int,
+        first try to convert it to int, if failed to convert, then log a
+        warning message and set the span_id to None.
+
+        :type span_id: int
+        :param span_id: Identifier for the span, unique within a trace.
+
+        :rtype: int
+        :returns: Span_id for the current span.
+        """
+        if not isinstance(span_id, int):
+            try:
+                span_id = int(span_id)
+            except (TypeError, ValueError):
+                logging.warning(
+                    'The type of span_id should be int, got {}.'.format(
+                        span_id.__class__.__name__))
+                span_id = None
+
+        return span_id
+
+    def check_trace_id(self, trace_id):
+        """Check the format of the trace_id to ensure it is 32-character hex 
+        value representing a 128-bit number.
+
+        :type trace_id: str
+        :param trace_id:
+        
+        :rtype: str
+        :returns: Trace_id for the current context.
+        """
+        trace_id_pattern = re.compile(_TRACE_CONTEXT_HEADER_FORMAT)
+
+        try:
+            match = trace_id_pattern.match(trace_id)
+
+            if match:
+                return trace_id
+            else:
+                logging.warning(
+                    'Trace_id {} does not the match the required format,'
+                    'generate a new one instead.'.format(trace_id))
+                return generate_trace_id()
+
+        except TypeError:
+            logging.warning(
+                'Trace_id should be str, got {}. Generate a new one.'.format(
+                    trace_id.__class__.__name__))
+            return generate_trace_id()
+
+    def __str__(self):
+        """Returns a string form of the TraceContext. This is the format of 
+        the Trace Context Header and should be forwarded to downstream
+        requests as the X-Cloud-Trace-Context header.
+
+        :rtype: str
+        :returns: String form of the TraceContext.
+        """
+        header = '{}/{};o={}'.format(
+            self.trace_id,
+            self.span_id,
+            self.enabled)
+        return header
+
+
+def generate_context_from_header(header):
+    """Parse a request header and generate a SpanContext object.
+
+    :type header: str
+    :param header: Value of a single HTTP request header from web application.
+
+    :rtype: :class:`~google.cloud.trace.trace_context.SpanContext`
+    :returns: A SpanContext object.
+    """
+    pattern = re.compile(_TRACE_CONTEXT_HEADER_FORMAT)
```

Comment:
Using regexp will be very expensive. Also we want to support multiple encoding formats:
* binary
* current google cloud format
* the new trace context format that we try to unify between zipkin, opencensus, etc. (https://github.com/TraceContext/tracecontext-spec/pull/1)

So maybe have a class:
cloud/google/trace/propagation/GoogleCloudFormat (with toHeader/fromHeader) that works with SpanContext.

Later we can add BinaryFormat that is used in gRPC (similar with https://github.com/census-instrumentation/opencensus-java/blob/master/impl_core/src/main/java/io/opencensus/trace/propagation/BinaryFormatImpl.java)